klasa kalkulatora

zyczynski opened the pull request.

zyczynski committed:
klasa kalkulatora

telemetrik commented on app/Money/Calculator/Amount.php:
```diff
@@ -0,0 +1,30 @@
+use Money\Currency;
+use Money\Money;
+
+class Amount
```
Jeżeli to jest kalkulator to dlaczego nazywa się "amount" ?

zyczynski replied:
nie wiem, chyba nie chciałem żeby podprzestrzeń nazywała się tak samo jak klasa, bo jeszce mi się to wszystko miesza

telemetrik replied:
Nazywaj rzeczy po funkcji, którą spełniają.

telemetrik commented on app/Money/Calculator/Amount.php:
```diff
@@ -0,0 +1,30 @@
+    public function getValue()
+    {
+
+        $this->value1=Money::PLN(800);
```
Dlaczego zapisujesz na sztywno wartości zamiast podawać ich z zewnątrz klasy? W ten sposób tym kalkulatorem mozesz obliczyc tylko sumę "80 zł + 80 zł". Równie dobrze klasa mogłaby się nazywać `Add80PLNTo80PLNCalculator`.

zyczynski committed:
poprawione

telemetrik commented on app/Money/Calculator/Calculator.php:
```diff
@@ -0,0 +1,30 @@
+private $value1;
+private $value2;
+
+    function __construct($value1,$value2)
```
Czemu w konstruktorze? Teraz żeby obliczyć dwie pary musisz tworzyć dwie instancje kalkulatora. Jaki jest cel żeby trzymać stan w kalkulatorze?

zyczynski replied:
a już myślałem :D

zyczynski replied:
Dwie instancje czyli dwa obiekty? Ja tu mam jeden i liczy....

telemetrik replied:
To zrób ćwiczenie - w pliku index dodaj dwa obliczenia - dla kwoty netto i brutto sumy 10 PLN + 49 PLN. Zobaczysz o co chodzi.

zyczynski replied:
No już wiem o co chodzi. oczywiście źle przeczytałem to co napisałeś i zgubiłem "pary"

zyczynski replied:
Nie no nie wiem, poddaje się. Wytłumacz mi o co chodzi albo coś podpowiedz bo mnie tu zaraz strzeli i krew nagła zaleje ;]

telemetrik replied:
Własności w klasie oznaczają, że klasa ma jakiś stan. Kalkukator służy tylko do obliczeń więc niepotrzebne mu $value1 i $value2 są niepotrzebne w klasie. Dostałeś młotek więc wszystko jest teraz dla Ciebie gwoździem. :) Dla udowodnienia, że to bez sensu zrób ćwiczenie - w pliku index dodaj dwa obliczenia - 10 PLN + 49 PLN i 20 PLN + 23 PLN. Będziesz musiał zrobić dwie instancje (new calculator) - po co? W prawdziwym życiu - kupujesz nowy kalkulator jak obliczysz jedno działanie?

zyczynski replied:
Tak, jak koncepcje rozumiem, ale za cholerę nie wiem jak to inaczej zapisać bezbłędnie ...;/

telemetrik replied:
No bez własności private $valueX w klasie.

telemetrik replied:
Kumasz o co chodzi teraz?

zyczynski replied:
chyba tak, zafiksowałem się na innym fragmencie i tak jak bym tego w ogóle nie widział ;/

telemetrik replied:
No to wytłumacz mi kiedy byś dodał własności w klasie a kiedy nie. :)

zyczynski replied:
Wtedy kiedy będzie to stała wartość, jak nazwisko, email, login etc. Jeśli zmienna to nie.

telemetrik replied:
Email możesz zmienić - to co z nim?

zyczynski replied:
Teoretycznie nazwisko też :) Nie no, myślę że powinien być dodany. Teoretycznie zmienić można, ale na pewno nie będzie zmieniany często. To są wartości przypisane do obiektu, jeden obiekt - jeden zestaw własności.

zyczynski committed:
Próbuje tak

w zapisie z metodą nic nie zmieniłem a mimo to wywala błąd "Call to a member function add() on null "

zyczynski committed:
ćwiczenie

zyczynski committed:
ufff

ale się z tym umordowałem ;(
